db/schema.rb

```diff
@@ -70,11 +70,12 @@
   end
 
   create_table "api_tokens", force: :cascade do |t|
-    t.bigint "lead_provider_id", null: false
+    t.bigint "lead_provider_id"
```

Comment:
So I'm not following here - I'm seeing a schema change to drop the `null: false` but no corresponding migration to achieve that? Is something missing here

You can also add a DB constraint in place of the `, null: false` requirement to require a `lead_provider_id IS NOT NULL WHERE scope = "lead_provider"`

Reply:
that is odd - not sure how that migration got lost - I'll add it back in.
Good idea about using a check constraint.